scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -155,6 +155,19 @@
       (write-status-to-datomic compute-cluster status)))
   {:cook-expected-state :cook-expected-state/completed})
 
+(defn handle-pod-externally-deleted
+  "Marks the corresponding job instance as failed in the database and
+  returns the `completed` cook expected state."
+  [{:keys [name] :as compute-cluster} pod-name]
+  (log/info "In compute cluster" name ", pod" pod-name "was externally deleted")
+  (when-not (api/synthetic-pod? pod-name)
+    (let [instance-id pod-name
+          status {:reason :reason-executor-terminated
```

Comment:
The reason string for this reason is "Mesos executor terminated". Let's use something less Mesos-y for this.

Reply:
I didn't find anything better without creating something

Reply:
Discussed offline, ended up creating a new reason